CMakeLists.txt

```
@@ -0,0 +1,59 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(SugarPP
+        VERSION 0.0.1
+        HOMEPAGE_URL "https://github.com/HO-COOH/SugarPP"
+        LANGUAGES CXX)
+
+# ---- Warning guard ----
+
+# Protect dependents from this project's warnings if the guard isn't disabled
+set(sugarpp_warning_guard "SYSTEM")
+if(SugarPP_INCLUDE_WITHOUT_SYSTEM)
+  set(sugarpp_warning_guard "")
+endif()
+
+# ---- Declare library ----
+
+add_library(SugarPP INTERFACE)
+add_library(SugarPP::SugarPP ALIAS SugarPP)
+
+target_include_directories(SugarPP
+        ${sugarpp_warning_guard}
+        INTERFACE
+        "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>")
+
+target_compile_features(SugarPP INTERFACE cxx_std_17)
+
+# ---- Install ----
+
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
+
+set(sugarpp_directory "sugarpp-${PROJECT_VERSION}")
+set(sugarpp_include_directory "${CMAKE_INSTALL_INCLUDEDIR}/${sugarpp_directory}")
+
+install(TARGETS SugarPP
+        EXPORT SugarPPTargets
+        INCLUDES DESTINATION "${sugarpp_include_directory}")
```

SugarCPP doesn't have any dependencies so this should just be
```suggestion
install(TARGETS SugarPP EXPORT SugarPPConfig)
```
Followed by
```cmake
install(EXPORT SugarPPConfig
        FILE SugarCPPConfig.cmake
        NAMESPACE SugarCPP::
        DESTINATION "${sugarpp_install_cmakedir}")
```

---

`sugarpp_include_directory` depends on `CMAKE_INSTALL_INCLUDEDIR` and that comes from `include(GNUInstallDirs)`, so it's nicer to keep things in one place.

While that gets rid of the additional file, I don't see that as a necessary change.

---

Nothing is being moved. Either renamed or deleted.

```patch
diff --git a/CMakeLists.txt b/CMakeLists.txt
index a181c13..90df913 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,7 +34,7 @@ set(sugarpp_directory "sugarpp-${PROJECT_VERSION}")
 set(sugarpp_include_directory "${CMAKE_INSTALL_INCLUDEDIR}/${sugarpp_directory}")
 
 install(TARGETS SugarPP
-        EXPORT SugarPPTargets
+        EXPORT SugarPPConfig
         INCLUDES DESTINATION "${sugarpp_include_directory}")
 
 set(sugarpp_install_cmakedir
@@ -46,12 +46,11 @@ write_basic_package_version_file(
         COMPATIBILITY SameMajorVersion
         ARCH_INDEPENDENT)
 
-install(EXPORT SugarPPTargets
+install(EXPORT SugarPPConfig
         NAMESPACE SugarPP::
         DESTINATION "${sugarpp_install_cmakedir}")
 
 install(FILES
-        "${PROJECT_SOURCE_DIR}/cmake/SugarPPConfig.cmake"
         "${PROJECT_BINARY_DIR}/SugarPPConfigVersion.cmake"
         DESTINATION "${sugarpp_install_cmakedir}")
 
diff --git a/cmake/SugarPPConfig.cmake b/cmake/SugarPPConfig.cmake
deleted file mode 100644
index b390cbc..0000000
--- a/cmake/SugarPPConfig.cmake
+++ /dev/null
@@ -1 +0,0 @@
-include("${CMAKE_CURRENT_LIST_DIR}/SugarPPTargets.cmake")
```

---

I guess this library will never need the extra things a separate config file can provide, so let's go with your suggestion.  
However, renaming the export set generated by `install(TARGETS ... EXPORT ...)` to `*Config` would just be misleading and pretty unconventional.